Src/IronPython/Runtime/PythonEncoding.cs

```diff
@@ -153,60 +169,131 @@ public override Decoder GetDecoder()
+            private Encoding _encoding;
+            private readonly Encoder _encoder;
+
+            public ProxyEncoder(Encoding encoding) {
+                _encoding = (Encoding)encoding.Clone();
+                _encoding.EncoderFallback = new ProxyEncoderFallback(_encoding.EncoderFallback.CreateFallbackBuffer(), _encoding.EncoderFallback.MaxCharCount);
+                _encoder = _encoding.GetEncoder();
+                Fallback = _encoder.Fallback = _encoding.EncoderFallback;
+            }
+
+            public override int GetByteCount(char[] chars, int index, int count, bool flush)
+                => _encoder.GetByteCount(chars, index, count, flush);
+
+            public override int GetBytes(char[] chars, int charIndex, int charCount, byte[] bytes, int byteIndex, bool flush)
+                => _encoder.GetBytes(chars, charIndex, charCount, bytes, byteIndex, flush);
+
+            public override void Reset() => _encoder.Reset();
+
+            private class ProxyEncoderFallback : EncoderFallback {
+                private readonly EncoderFallbackBuffer _buffer;
+                private readonly int _maxCharCount;
+
+                public ProxyEncoderFallback(EncoderFallbackBuffer buffer, int maxCharCount) {
+                    _buffer = buffer;
+                    _maxCharCount = maxCharCount;
+                }
+
+                public override EncoderFallbackBuffer CreateFallbackBuffer() => _buffer;
+                public override int MaxCharCount => _maxCharCount;
+            }
+        }
+
         public class PythonEncoder : Encoder {
             private readonly PythonEncoding _parentEncoding;
             private readonly Encoder _pass1encoder;
             private Encoder _pass2encoder;
-            private readonly int _characterWidth;
 
             public PythonEncoder(PythonEncoding parentEncoding) {
                 _parentEncoding = parentEncoding;
-                _characterWidth = _parentEncoding.CharacterWidth;
+                _pass1encoder = parentEncoding.Pass1Encoding.GetEncoder();
+
+                if (!(_pass1encoder.Fallback is PythonEncoderFallback) && _parentEncoding.Pass1Encoding.EncoderFallback is PythonEncoderFallback) {
+                    // Non-conformant Encoder implementation, the challenge is to get to the fallback buffer used by such encoder.
+
+                    // Possibility 1: _pass1encoder is EncoderNLS .
+                    // This weirdo (.NET Core only) doe not use Falback and FallbackBuffer properties from its Encoder base class;
+                    // it redefines them as new properties and uses them instead.
+                    // Although the new FallbackBuffer is public, it is not easilly accessible because the EncoderNLS class is internal.
+                    // One way of accessing it is by reflection. This will be handled by GetPythonEncoderFallbackBuffer()
+                    if (_pass1encoder.GetType().FullName == "System.Text.EncoderNLS") return;
```

Comment:
If they do this, than yes, it will be a whole new game. But if it happens, tests in `EncodingTests.cs` will start failing.

Frankly I really detest this particular workaround, probably mostly because the breaking change that .NET Core introduced seems so senseless and unnecessary. To be more specific, I don't know the reason why they did it, but for any possible benefit I can think of that this change provides, there is another way to achieve it without regressing on functionality. I am considering opening an Issue in the corefx repo.

Reply:
Presumably there are technical reasons for why they did it the way they did (for example wanting to have the System.Text.Encoding.CodePages package work on .NET Framework). In any case, you should definitely open an issue in the dotnet/runtime repo. At the very least they might explain the reasoning...